From a0aa6d6ec949e04fcd0883610a24e163d81a40e3 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Sat, 16 Jun 2018 21:09:43 +0200
Subject: [PATCH 09/10] TrustInterface: Handle unsupported encryption state

 * Some devices might return unsupported, therefore
    explicitly label this state as bad

 * Also applies for special situations where encryption
    is essentially impossible, like booting with
    MultiROM from an external storage

Change-Id: I7752933bad05bf8c67f5ca0d3fba7c5b38833c87
---
 .../org/lineageos/platform/internal/TrustInterfaceService.java  | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/TrustInterfaceService.java b/lineage/lib/main/java/org/lineageos/platform/internal/TrustInterfaceService.java
index e95f395..cc1db10 100644
--- a/lineage/lib/main/java/org/lineageos/platform/internal/TrustInterfaceService.java
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/TrustInterfaceService.java
@@ -312,6 +312,8 @@ public class TrustInterfaceService extends LineageSystemService {
                 return isOldDevice ?
                         TrustInterface.TRUST_FEATURE_LEVEL_POOR :
                         TrustInterface.TRUST_FEATURE_LEVEL_BAD;
+            case DevicePolicyManager.ENCRYPTION_STATUS_UNSUPPORTED:
+                return TrustInterface.TRUST_FEATURE_LEVEL_BAD;
             default:
                 return TrustInterface.ERROR_UNDEFINED;
         }
-- 
2.17.1

